src/payg/payg.sync.service.ts

```diff
@@ -0,0 +1,134 @@
+// Copyright 2020-2022 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { ApolloClient } from '@apollo/client/core';
+import { Injectable, OnApplicationBootstrap } from '@nestjs/common';
+import { GraphqlQueryClient, NETWORK_CONFIGS } from '@subql/network-clients';
+import { GetStateChannels, GetStateChannelsQuery } from '@subql/network-query';
+
+import { BigNumber, utils } from 'ethers';
+
+import { Config } from '../configure/configure.module';
+import { ContractService } from '../core/contract.service';
+import { getLogger } from '../utils/logger';
+import { PaygService } from './payg.service';
+
+const logger = getLogger('pagy');
+
+@Injectable()
+export class PaygSyncService implements OnApplicationBootstrap {
+  private client: ApolloClient<unknown>;
+
+  constructor(
+    private contractService: ContractService,
+    private paygServicee: PaygService,
+    private readonly config: Config,
+  ) {
+    const queryClient = new GraphqlQueryClient(NETWORK_CONFIGS[config.network]);
+    this.client = queryClient.networkClient;
+  }
+
+  onApplicationBootstrap() {
+    void (() => {
+      this.syncStateChannelsPeriodically();
+      this.subscribeStateChannelEvents();
+    })();
+  }
+
+  syncStateChannelsPeriodically() {
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    setInterval(async (): Promise<void> => {
+      try {
+        logger.debug(`load from Subquery Project...`);
+        const result = await this.client.query<GetStateChannelsQuery>({
+          // @ts-ignore
+          query: GetStateChannels,
+          variables: { status: 'OPEN' },
+        });
+
+        await Promise.all(
+          result.data.stateChannels.nodes.map((stateChannel) =>
+            this.paygServicee.syncChannel(
+              BigNumber.from(stateChannel.id).toString(),
+              stateChannel.deployment.id,
+              stateChannel.indexer,
+              stateChannel.consumer,
+              stateChannel.agent,
+              stateChannel.total.toString(),
+              stateChannel.spent.toString(),
+              stateChannel.price.toString(),
+              new Date(stateChannel.expiredAt).valueOf() / 1000,
+              new Date(stateChannel.terminatedAt).valueOf() / 1000,
+              stateChannel.terminateByIndexer,
+              stateChannel.isFinal,
+            ),
```

Comment:
catch can be added here so one failure won't fail the whole batch, also that we can have more accurate errors
```suggestion
            ).catch(e=>logger.error(`Failed to sync state channels ${stateChannel.id} from Subquery Project: ${e}`)),
```